src/components/molecules/VenueJoinButton/VenueJoinButton.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useCallback } from "react";
+import { VenueAccessMode } from "types/VenueAcccess";
+import { AnyVenue } from "types/venues";
+import { getCurrentTimeInUTCSeconds, getTimeBeforeParty } from "utils/time";
+import { joinVenue } from "utils/url";
+import { SecretPasswordForm } from "../SecretPasswordForm";
+
+interface VenueJoinButtonProps {
+  venueId?: string;
+  venue: AnyVenue;
+  onPasswordSubmit: () => void;
+  onPasswordSuccess: () => void;
+}
+
+export const VenueJoinButton: React.FC<VenueJoinButtonProps> = ({
+  venueId,
+  venue,
+  onPasswordSubmit,
+  onPasswordSuccess,
+}) => {
+  const handleJoinVenue = useCallback(() => {
+    joinVenue(venueId, venue?.entrance);
+  }, [venue, venueId]);
+
+  // @debt Handle emails and codes as well.
+  if (venue.access === VenueAccessMode.Password) {
+    return (
+      <div className="secret-password-form-wrapper">
+        <SecretPasswordForm
+          buttonText={
+            venue.config?.landingPageConfig.joinButtonText ?? "Join the party"
+          }
+          onPasswordSubmit={onPasswordSubmit}
+          onPasswordSuccess={onPasswordSuccess}
+        />
+      </div>
+    );
+  }
+
+  return (
+    <button
+      className="btn btn-primary btn-block btn-centered"
+      onClick={handleJoinVenue}
+    >
```

Comment:
This will always show the secret password form if that mode is configured, so the JSX below won't be rendered. Is that the desired state/logic flow here?

Reply:
Yes, this was a `renderFunction` with switch case returning different components. Now it's a component with the intention to add `codes` and `emails` with different button or UI

Reply:
Cool, makes sense :)